ingest/src/test/java/com/microsoft/azure/kusto/ingest/IngestClientImplTest.java

```diff
@@ -171,89 +181,29 @@ void IngestFromStream_NullStreamSourceInfo_IllegalArgumentException() {
-                Paths.get(System.getProperty("java.io.tmpdir"), String.valueOf(System.currentTimeMillis())).toString());
-        //noinspection ResultOfMethodCallIgnored
-        tempPath.mkdirs();
-
-        String tempFolderPath = tempPath.toString();
-
-        ingestClientSpy.ingestFromResultSet(resultSetSourceInfo, ingestionProperties, tempFolderPath);
-        verify(ingestClientSpy).ingestFromResultSet(resultSetSourceInfo, ingestionProperties, tempFolderPath);
-
-        // captor to allow us to inspect the internal call values
-        ArgumentCaptor<FileSourceInfo> fileSourceInfoCaptor = ArgumentCaptor.forClass(FileSourceInfo.class);
-        verify(ingestClientSpy).ingestFromFile(fileSourceInfoCaptor.capture(), any(IngestionProperties.class));
-        FileSourceInfo fileSourceInfoActual = fileSourceInfoCaptor.getValue();
-        assertEquals(numberOfChars * 2, fileSourceInfoActual.getRawSizeInBytes());
-        // make sure the temp file was written to the folder we specified
-        Assertions.assertTrue(fileSourceInfoActual.getFilePath().startsWith(tempFolderPath));
+    void IngestFromResultSet_NullIngestionProperties_IllegalArgumentException() {
+        ResultSetSourceInfo resultSetSourceInfo = mock(ResultSetSourceInfo.class);
+        assertThrows(
+                IllegalArgumentException.class,
+                () -> ingestClientImpl.ingestFromResultSet(resultSetSourceInfo, null));
     }
 
     @Test
-    void IngestFromResultSet_ResultSetToCsv_IngestionClientException()
-            throws IngestionClientException, IngestionServiceException {
-        IngestClientImpl ingestClient = new IngestClientImpl(resourceManagerMock, azureStorageClientMock);
-        // we need a spy to intercept the call to ingestFromFile so it wouldn't be called
-        IngestClientImpl ingestClientSpy = spy(ingestClient);
-        TableReportIngestionResult ingestionResultMock = mock(TableReportIngestionResult.class);
-
-        doReturn(ingestionResultMock).when(ingestClientSpy).ingestFromFile(any(), any());
-        doThrow(new IngestionClientException("error in resultSetToCsv"))
-                .when(ingestClientSpy)
-                .writeResultSetToWriterAsCsv(any(), any(), anyBoolean());
-
-        ResultSetSourceInfo resultSetSourceInfo = new ResultSetSourceInfo(mock(ResultSet.class));
-
+    void IngestFromResultSet_NullResultSetSourceInfo_IllegalArgumentException() {
         assertThrows(
-                IngestionClientException.class,
-                () -> ingestClientSpy.ingestFromResultSet(resultSetSourceInfo, ingestionProperties));
+                IllegalArgumentException.class,
+                () -> ingestClientImpl.ingestFromResultSet(null, ingestionProperties));
     }
 
     @Test
-    void IngestFromResultSet_FileIngest_IngestionClientException()
-            throws IngestionClientException, IngestionServiceException, SQLException {
+    void IngestFromResultSet_StreamIngest_IngestionClientException() throws Exception {
```

Comment:
why did you remove the old test?

Reply:
It is the same test, the old implementation use temp storage to convert ResultSet to CSV and then calls IngestFromFile. In the new implementation the library reads the ResultSet in memory and writes it to stream so the method calls IngestFromStream instead.